lib/routemaster/responses/hateoas_response.rb

```diff
@@ -24,28 +45,95 @@ def method_missing(m, *args, &block)
 
         if _links.keys.include?(normalized_method_name)
           unless respond_to?(method_name)
-            resource = Resources::RestResource.new(_links[normalized_method_name]['href'], client: @client)
-
             self.class.send(:define_method, method_name) do |*m_args|
-              resource
+              build_resource(normalized_method_name)
             end
-
-            resource
           end
+          self.send(method_name)
         else
           super
         end
       end
 
       def body_without_links
-        body.reject { |key, _| ['_links'].include?(key) }
+        body.tap { |b| b.delete('_links') }
       end
 
       private
 
+      def paginated_and_first_page?
+        @response.body.has_key?('page') && @response.body['page'] == 1
+      end
+
+      def build_resource(resource_name)
+        resource = _links[resource_name]
+        if resource.is_a? Hash
+          build_resource_from_href(resource['href'])
+        else
+          # Must be an Array
+          if paginated_and_first_page?
+            lazy_list_of_resources_in_pages(resource_name, resource)
```

Comment:
you're passing the `resource_name` here as an argument but the method definition asks for a generic `method_name`, can we change either one or the other for consistency and clarity?